### ISSUE TITLE
Using Cellrank for metabolic-labelling-based RNA velocity

### ISSUE THREAD

**Issue by 117999** [theislab/cellrank#811]
...
Hi Marius, 

I am wondering whether CellRank can be directly used for the metabolic-labeling experiments, and the labeling RNA velocity computed with Dynamo, which is released recently on Cell. 

Thanks

**Comment by Marius1311** [comment]
Hi @117999, thanks for your question. As long as dynamo gives you high-dimensional velocity vectors, this should work by simply applying the `VelocityKernel` to it. This is quite an interesting application, so please keep us up to date with your experiments and let us know in case you encounter difficulties along the way.

**Comment by 117999** [comment]
Hi Marius, 

Thanks a lot for your timely reply. I have tried to apply the VelocityKernel and compute the transition matrix. However, warnings appeared that `WARNING: KNN graph is not symmetric `
and upon computing the transition matrix, it shows that `ValueError: Matrix is not row-stochastic. The following rows don't sum to 1`

I am wondering why would the KNN graph not symmetric, and what's the interpretation for my data?

Thank you

**Comment by Marius1311** [comment]
How did you compute the KNN graph?

**Comment by Marius1311** [comment]
(in my analogy, x and y are cells)

**Comment by Marius1311** [comment]
Hi @117999, did my post above help in any way?

**Comment by 117999** [comment]
Hi Marius,

Sorry for my late response. I have followed your suggestions to use the connectivities. It appeared that this works well for the expression profile. However, the problem arises when I tried to compute the transition matrix from the VelocityKernel.

Here follows my code.
`sc.pp.neighbors(adata)
vk_n = VelocityKernel(adata,vkey='velocity_T',xkey='X_total')
print(vk_n)
vk_n.compute_transition_matrix()`
It turns out that `ValueError: Matrix is not row-stochastic. The following rows don't sum to 1: `[nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan nan`
